### ISSUE TITLE
failed collecting os metrics

### ISSUE THREAD

**Issue by xll970211** [prometheus-community/windows_exporter#923]
i run win11 and windows server 2008 is ok,but run in windows server 2012 has error
```
time="2022-01-21T15:48:57+08:00" level=error msg="Error reading textfile collect
or directory \"C:\\\\Users\\\\Administrator\\\\Desktop\\\\textfile_inputs\": ope
n C:\\Users\\Administrator\\Desktop\\textfile_inputs: The system cannot find the
 file specified." source="textfile.go:249"
time="2022-01-21T15:48:57+08:00" level=error msg="failed collecting os metrics:<
nil> The system cannot find the file specified." source="os.go:137"
time="2022-01-21T15:48:57+08:00" level=error msg="collector os failed after 0.00
0980s: The system cannot find the file specified." source="exporter.go:207"
time="2022-01-21T15:49:02+08:00" level=error msg="Error reading textfile collect
or directory \"C:\\\\Users\\\\Administrator\\\\Desktop\\\\textfile_inputs\": ope
n C:\\Users\\Administrator\\Desktop\\textfile_inputs: The system cannot find the
 file specified." source="textfile.go:249"
time="2022-01-21T15:49:02+08:00" level=error msg="failed collecting os metrics:<
nil> The system cannot find the file specified." source="os.go:137"
time="2022-01-21T15:49:02+08:00" level=error msg="collector os failed after 0.00
1958s: The system cannot find the file specified." source="exporter.go:207"
```

**Comment by yuanpengchao** [comment]
you need run mkidr `textfile_inputs` at C:\Users\Administrator\Desktop\

Our Windows English server is running properly, Chinese server also has this error:
failed collecting os metrics:<
nil> The system cannot find the file specified." source="os.go:137

**Comment by xll970211** [comment]
thank you very much.i I tried what you said mkdir textfile_inputs. The result is the same as you said
failed collecting os metrics:<
nil> The system cannot find the file specified." source="os.go:137
but we run in windows server 2008R2 Chinese，It won't happen
What should I do? @yuanpengchao

**Comment by yuanpengchao** [comment]
@xll970211  I'm looking for the answer to that question, too ……囧

**Comment by csrlsm** [comment]
I use release 0.171 and the OS windows server 2012 , there is the same issue. 
but in windows server 2019 it works

> i run win11 and windows server 2008 is ok,but run in windows server 2012 has error
> 
> ```
> time="2022-01-21T15:48:57+08:00" level=error msg="Error reading textfile collect
> or directory \"C:\\\\Users\\\\Administrator\\\\Desktop\\\\textfile_inputs\": ope
> n C:\\Users\\Administrator\\Desktop\\textfile_inputs: The system cannot find the
>  file specified." source="textfile.go:249"
> time="2022-01-21T15:48:57+08:00" level=error msg="failed collecting os metrics:<
> nil> The system cannot find the file specified." source="os.go:137"
> time="2022-01-21T15:48:57+08:00" level=error msg="collector os failed after 0.00
> 0980s: The system cannot find the file specified." source="exporter.go:207"
> time="2022-01-21T15:49:02+08:00" level=error msg="Error reading textfile collect
> or directory \"C:\\\\Users\\\\Administrator\\\\Desktop\\\\textfile_inputs\": ope
> n C:\\Users\\Administrator\\Desktop\\textfile_inputs: The system cannot find the
>  file specified." source="textfile.go:249"
> time="2022-01-21T15:49:02+08:00" level=error msg="failed collecting os metrics:<
> nil> The system cannot find the file specified." source="os.go:137"
> time="2022-01-21T15:49:02+08:00" level=error msg="collector os failed after 0.00
> 1958s: The system cannot find the file specified." source="exporter.go:207"
> ```

**Comment by breed808** [comment]
The text file directory isn't relevant to the `os` collector, as it doesn't read the file.

However, it does open a few registry keys: `SYSTEM\CurrentControlSet\Control\Session Manager\Memory Management` and `SOFTWARE\Microsoft\Windows NT\CurrentVersion`.

Are these present on the affected systems? Is this issue affecting all systems in a consistent manner?

**Comment by xll970211** [comment]
v1.18.1 in windows server 2012R2 is ok

**Comment by breed808** [comment]
The page file issue will be resolved in #954. Once that's merged, let me know if the issue persists with the `os` collector and we can investigate further.

**Comment by petemounce** [comment]
Thanks very much!
